### DISCUSSION TITLE
Airflow UI  404 - page not found

### DISCUSSION THREAD

**Discussion by boukisge** [apache/airflow#20524] [General]
### Apache Airflow version

2.2.1

### What happened

Unable to deploy Apache Airflow 2.2.x in Enterprise OpenShift 4.x cluster . I get a blank page in chrome browserAirflow UI  404 - page not found

for http://"myairflow-web-OpenShift route url"/airflow

### What you expected to happen

Not permitted to create RBAC roles and service accounts in our enterprise OpenShift cluster. Connecting to external postgreSQL database and redis container not an issue

The following pods are up and running
db-migations
flower
scheduler
web
worker
redis server

There is linkage between route, service and pod for flower and web
flower UI comes up but no html 

### How to reproduce

Deploy Helm Community version to OpenShift 4.4.x

### Operating System

Debian GNU/Linux v10

### Deployment

Official Apache Airflow Helm Chart

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
There is not  nearly enough details that anyone will be able to help you here, I am afraid - you need to provide more details and reproducible  path to classify it as Airflow bug. feel free to continue discussion, provide logs and possibly even discuss in slack in #troubleshooting session however.

**Reply by boukisge** [reply]
How can I configure Apache Airflow UI to work without the need to port-forward

**Reply by boukisge** [reply]
Apache Airflow is up and running , but only when I port-forward using the following command : kubectl port-forward svc/airflow-web:8080:8080 --namespace <my namespace>   and launching the UI via http://localhost:8080 . Given that the web container has been exposed via a Route/Ingress , I should be able to access the UI via the route: http://airflow-webserver-"os route url"/airflow

**Comment by boukisge** [comment]
Here is some additional info
Kubernetes Ingress endpoint URL has been created , but I get a 404 UI page error not found
Airflow Webserver: http://airflow-webserver.cp-7196860.<cluster url api>/airflow

When I execute the following port-forward command , The Airflow UI does come up on my local host chrome browser
oc port-forward svc/airflow-web 8080:8080 --namespace cp-7196860

**Reply by potiuk** [reply]
You should check in your webserver logs and see what URL you are accessing on the webserver.
 
But I assume you do not do to the "/airlfow" mapping properly. You should reverse-proxy the URLS if you want to use /airflow 

Also what might actually be needed is the https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#enable-proxy-fix set to True.

**Reply by boukisge** [reply]
AIRFLOW__WEBSERVER_BASE_URL: "http://"Route URL"/airflow  - this is standard url that works for other containers running on the K8s/OS cluster

**Reply by boukisge** [reply]
I will take a look at the enable proxy instructions, Thx

**Reply by potiuk** [reply]
AIRFLOW__WEBSERVER_BASE_URL does not do what you think it does:

From: https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#webserver

> The base url of your website as airflow cannot guess what domain or cname you are using. This is used in automated emails that airflow sends to point links to the right web server

**Reply by potiuk** [reply]
In short -> requests to Airflow webserver should always come without "/airflow" path. The "/airflow" path needs to be stripped-off and handled properly with your ingress to do URL rewrite.

**Comment by boukisge** [comment]
How do I configure this in values yank. Do you have an example that you can
share with me?

**Reply by potiuk** [reply]
No idea to be honest. But  you can likely find it in docs that our docs point to:

https://airflow.apache.org/docs/helm-chart/stable/production-guide.html#accessing-the-airflow-ui

> Ingress
> You can create and configure Ingress objects. See the Ingress chart parameters. For more information on Ingress, see the Kubernetes > Ingress documentation.

**Comment by boukisge** [comment]
Thx - I have enough info to move forward

**Reply by potiuk** [reply]
Likely you need to configure rewrite-target annotation: https://github.com/kubernetes/ingress-nginx/blob/main/docs/examples/rewrite/README.md (after few seconds of looking at it - so do not take my word for it).

**Comment by boukisge** [comment]
Issue resolved 
Solution:
Modify  the value of environment variable: AIRFLOW__WEBSERVER__BASE_URL 
From: http://<application-namespace-cluster_route.com/airflow
To: http://localhost:8080/airflow

**Reply by potiuk** [reply]
Interesting . Anything else ? Or just this ?

**Comment by boukisge** [comment]
A few minor changes in ingress.web and ingress.flower but that is it

**Reply by tanmaykansara** [reply]
Hey 
I am running into a similar problem. The diff is that when I set base_url to localhost:8080/test the pod does not spin up(keeps restarting). I assume that the /health is returning a 404. 

Anything special that you did for this ?

```
extraEnv: |
  - name: "AIRFLOW_VAR_FOO"
    value: "Develp_foo"
  - name: "AIRFLOW__WEBSERVER__BASE_URL"
    value: "http://localhost:8080/test"
executor: "KubernetesExecutor"
webserverSecretKeySecretName: dev-webserver-secret
ingress:
  enabled: true
  
 
  apiVersion: networking.k8s.io/v1
  
  ## airflow webserver ingress configs
  web:
    pathType: "Prefix"
    path: "/test"
    host: "k8s-dev-airflowi-c33***30-1**********.us-west-2.elb.amazonaws.com"  
```



Above is how my config is. I know ingress is working as the http://host/test results in 404 airflow page. 
Any help will be appreciated.  
Thanks
